x-pack/platform/plugins/shared/onechat/server/services/data/data_service.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+import {
+  createDataTypeRegistry,
+  type DataTypeRegistry,
+  type DataTypeDefinition,
+} from './data_type_registry';
+
+export interface DataServiceSetupDeps {
+  logger: Logger;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface DataServiceStartDeps {}
+
+export interface DataServiceSetup {
+  register(dataType: DataTypeDefinition): void;
+}
+
+export interface DataServiceStart {
+  getRegistry(): Promise<DataTypeRegistry>;
+}
+
+export class DataService {
```

Comment:
This sounds too generic for me, in a way ES is "data service" :D we should have something like DataTypeService / DataTypeRegistry here